marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -0,0 +1,314 @@
+import Data.List.NonEmpty (nonEmpty, toList)
+import Data.Maybe (fromJust, isJust, mapMaybe)
+import Data.Proxy (Proxy (Proxy))
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Database.SQLite.Simple qualified as SQL
+import Hedgehog (Property, cover, forAll, property, (===))
+import Hedgehog qualified
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Hedgehog (testPropertyNamed)
+
+import Cardano.Api qualified as C
+import Control.Monad (void)
+import Data.List qualified as List
+import Gen.Cardano.Api.Typed qualified as CGen
+import Gen.Marconi.ChainIndex.Indexers.Utxo (genEventAtChainPoint, genUtxoEvents)
+import Marconi.ChainIndex.Indexers.Utxo (StorableEvent (ueInputs, ueUtxos))
+import Marconi.ChainIndex.Indexers.Utxo qualified as Utxo
+import Marconi.ChainIndex.Types (CurrentEra, TargetAddresses)
+import Marconi.Core.Storable (StorableQuery, resume)
+import Marconi.Core.Storable qualified as Storable
+
+-- | Proves two list are equivalant, but not identical
+
+-- NOTE --
+-- | UtxoEvents equivalent relationship
+-- Not all utxoEvent attributes have defined `Eq` and/or `Ord` relationship defined.
+-- As events are disassembled and reassembled, the Ordering of these sub-parts may change in the coresponding collections.
+-- Therefore we used the Equivalence relationship to show two event are morally equal.
+equivalentLists :: Eq a => [a] -> [a] -> Bool
+equivalentLists us us' =
+  length us == length us'
+  &&
+  all (const True) [u `elem` us'| u <- us]
+  &&
+  all (const True) [u `elem` us| u <- us']
+
+tests :: TestTree
+tests = testGroup "Spec.Marconi.ChainIndex.Indexers.Utxo"
+    [ testPropertyNamed
+        "filter UtxoEvent for Utxos with address in the TargetAddress"
+        "eventsAtAddressTest"
+        eventsAtAddressTest
+
+    , testPropertyNamed
+        "marconi-utxo event-to-sqlRows property"
+        "eventsToRowsRoundTripTest"
+        eventsToRowsRoundTripTest
+
+    , testPropertyNamed
+        "marconi-utxo storable-query address property"
+        "txAddressToUtxoAddressTest"
+        txAddressToUtxoAddressTest
+
+    , testPropertyNamed
+        "marconi-utxo storage-roundtrip property"
+        "utxoStorageTest"
+        utxoStorageTest
+
+    -- TODO BROKEN
+    -- , testPropertyNamed
+    --     "marconi-utxo insert-query property"
+    --     "utxoInsertAndQueryTest"
+    --     utxoInsertAndQueryTest
+
+    , testPropertyNamed
+        "marconi-utxo rewind property"
+        "rewindTest"
+        rewindTest
+
+    , testPropertyNamed
+        "marconi-utxo query-interval property"
+        "utxoQueryIntervalTest"
+        utxoQueryIntervalTest
+
+    , testPropertyNamed
+          "The points that indexer can be resumed from should return at least the genesis point"
+          "propResumingShouldReturnAtLeastTheGenesisPoint"
+          propResumingShouldReturnAtLeastTheGenesisPoint
```

Comment:
(Not sure if this is the case currently but) If one already knows the list of points resume is supposed to return, then doing `expectedPoints === gottenPoints` would compare both the content and the order in a single assertion. Otherwise agree that testing multiple models in a single test is confusing.

A related thought on resume, what if it would just return the optimal point to resume from, something like: from storage find *only* the latest event that is past the security parameter (and otherwise genesis)? Because if you always return both then during the query to node to find the block to continue syncing from, it wouldn't report an error if that latest event is not found and start from genesis instead. But you would want to know when your assumption about the security parameter is broken, right?